packages/react/src/components/Tooltip/Tooltip.tsx

```diff
@@ -1,138 +1,119 @@
-import { FC, useState, useRef } from "react";
+import React, { FC, useState, useRef } from "react";
 import classNames from "classnames";
 import useGlobalSettings from "../../hooks/useGlobalSettings";
 import { TooltipProps } from "./Tooltip.props";
-import ReactDOM from "react-dom";
+import { createPopper, Instance as PopperInstance } from "@popperjs/core";
 
 const Tooltip: FC<TooltipProps> = ({
   className,
   children,
   icon,
   label,
+  iconTheme,
   theme,
   id,
 }) => {
   const { prefix } = useGlobalSettings();
   const baseClass = `${prefix}--tooltip`;
   const [isVisible, setIsVisible] = useState<boolean>(false);
-  const [position, setPosition] = useState<{ x: number; y: number }>({
-    x: 0,
-    y: 0,
-  });
-  const [arrowPlacement, setArrowPlacement] = useState<string>("center");
-  const [arrowAlignment, setArrowAlignment] = useState<string>("left");
+  const [longTooltip, setLongToolTip] = useState<boolean>(false);
+  const [popperInstance, setPopperInstance] = useState<PopperInstance | null>(
```

Comment:
@Shashika6 & @bashlk are we sure we want to do this with state? Would it not be better to use a ref to store the instance?

Reply:
Yeah this would be better as a ref since the popper instance is mutable ref.

Reply:
@bashlk I noticed that we can just use a function to check on the fly. Something like 

```suggestion
  const isLongTooltip = () => {
    const tooltipText = (tooltipRef.current?.textContent || tooltipRef.current?.innerText || '').trim();
    return tooltipText.length > 50;
  };
```

What do you think ?

Reply:
Yeah that could be alright - my only concern is that the ref could be undefined while this is evaluates by I guess this runs onMouseOver when the ref has already been mounted.

Reply:
Yup it runs on mouseOver so in that case will be removing the state